src/Confluent.Kafka/AdminClient.cs

```diff
@@ -0,0 +1,750 @@
+            { Librdkafka.EventType.DescribeConfigs_Result, typeof(TaskCompletionSource<List<DescribeConfigResult>>) },
+            { Librdkafka.EventType.AlterConfigs_Result, typeof(TaskCompletionSource<List<AlterConfigResult>>) },
+            { Librdkafka.EventType.CreatePartitions_Result, typeof(TaskCompletionSource<List<CreatePartitionsResult>>) }
+        };
+
+        /// <summary>
+        ///     Get the configuration for the specified resources. The returned 
+        ///     configuration includes default values and the IsDefault property
+        ///     can be used to distinguish them from user supplied values. The 
+        ///     value of config entries where IsSensitive is true is always null 
+        ///     so that sensitive information is not disclosed. Config entries where
+        ///     IsReadOnly is true cannot be updated. This operation is supported 
+        ///     by brokers with version 0.11.0.0 or higher.
+        /// </summary>
+        /// <param name="resources">
+        ///     The resources (topic and broker resource types are currently 
+        ///     supported)
+        /// </param>
+        /// <param name="options">
+        ///     The options to use when describing configs.
+        /// </param>
+        /// <returns>
+        ///     Configs for the specified resources.
+        /// </returns>
+        public Task<List<DescribeConfigResult>> DescribeConfigsAsync(IEnumerable<ConfigResource> resources, DescribeConfigsOptions options = null)
+        {
+            // TODO: To support results that may complete at different times, we may also want to implement:
+            // List<Task<DescribeConfigResult>> DescribeConfigsConcurrent(IEnumerable<ConfigResource> resources, DescribeConfigsOptions options = null)
+
+            var completionSource = new TaskCompletionSource<List<DescribeConfigResult>>();
+            var gch = GCHandle.Alloc(completionSource);
+            Handle.LibrdkafkaHandle.DescribeConfigs(
+                resources, options, resultQueue,
+                GCHandle.ToIntPtr(gch));
+            return completionSource.Task;
+        }
+
+        /// <summary>
+        ///     Update the configuration for the specified resources. Updates are not transactional
+        ///     so they may succeed for some resources while fail for others. The configs for a 
+        ///     particular resource are updated atomically. This operation is supported by brokers 
+        ///     with version 0.11.0.0 or higher.
+        /// </summary>
+        /// <param name="configs">
+        ///     The resources with their configs (topic is the only resource type with configs
+        ///     that can be updated currently)
+        /// </param>
+        /// <param name="options">
+        ///     The options to use when altering configs.
+        /// </param>
+        /// <returns>
+        ///     The results of the alter configs requests.
```

Comment:
... and this is exposed via an exception (rather than in the return value).

Reply:
there's a dilemma here. The return values include an `Error` field, but this is always set to `NoError` (so it's misleading that it exists). but having the `Error` property means the type can be re-used by the corresponding exception type, which gets thrown when any results are in error. Also, it means we naturally have an extensible container type, whereas otherwise we'd have just string or something. Anyway, still considering the return type here. Probably won't block this PR on this.